core/src/main/java/ai/timefold/solver/core/impl/domain/variable/ShadowVariableUpdateHelper.java

```diff
@@ -84,6 +84,7 @@ public void updateShadowVariables(Solution_ solution) {
                 .stream()
                 .map(Object::getClass)
                 .distinct()
+                .filter(clazz -> clazz.isAnnotationPresent(PlanningEntity.class))
                 .toArray(Class[]::new);
```

Comment:
Using `SolutionDescriptor.visitAllEntities()` will minimize iteration and reuse logic which already exists. The result should be the exact same, because `getAllEntitiesAndProblemFacts()` calls `visitAllEntities()` under the hood anyway.

Ideally we'd iterate just the entity descriptors, but I realize that - at this point - they might not yet exist.